tests/diff/index.c

```diff
@@ -239,3 +239,34 @@ void test_diff_index__reports_conflicts_when_reversed(void)
 	cl_assert_equal_i(2, exp.line_adds);
 	cl_assert_equal_i(5, exp.line_dels);
 }
+
+void test_diff_index__not_in_head_conflicted(void)
+{
+	const char *a_commit = "26a125ee1bf"; /* the current HEAD */
+	git_index_entry ancestor = {{0}}, ours = {{0}}, theirs = {{0}};
+	git_index *index;
+	git_diff *diff;
+	const git_diff_delta *delta;
+
+	git_tree *a = resolve_commit_oid_to_tree(g_repo, a_commit);
+
+	cl_git_pass(git_repository_index(&index, g_repo));
+	cl_git_pass(git_index_read_tree(index, a));
+
+	ancestor.path = ours.path = theirs.path = "file_not_in_head";
+	ancestor.mode = ours.mode = theirs.mode = 0100644;
+	git_oid_fromstr(&ancestor.id, "deadbeefdeadbeefdeadbeefdeadbeefdeadbeef");
+	git_oid_fromstr(&ours.id, "deadbeefdeadbeefdeadbeefdeadbeefdeadbeef");
+	git_oid_fromstr(&theirs.id, "deadbeefdeadbeefdeadbeefdeadbeefdeadbeef");
+	cl_git_pass(git_index_conflict_add(index, &ancestor, &ours, &theirs));
```

Comment:
If the file does not exist in HEAD, then there won't be an 'ours' entry in the conflict, and possibly not one in 'ancestor' either.